src/Swarm/Game/Entity.hs

```diff
@@ -107,7 +108,8 @@ import GHC.Generics (Generic)
 import Swarm.Game.Display
 import Swarm.Game.Location
 import Swarm.Language.Capability
-import Swarm.Util (binTuples, dataNotFound, getDataFileNameSafe, plural, reflow, (?))
+import Swarm.TUI.Model.FailureRender (prettyPathLoadingFailure)
```

Comment:
@kostmo please move the failure type (not widget render) that game needs to its directory.

If `Game` needs to load something and pretty print it in text, then it should be in its directory.

I understand it's less convenient, but at least we will not be lumping all errors into one big pile.

Reply:
I don't quite follow what the proper resolution here is; can you be more explicit? :)

Perhaps the convention will become more clear once #1043 is merged.

Reply:
@kostmo imagine that you could not import TUI from Game and write code accordingly. 😉 

If you decide to ignore it then hope I will not merge the library split faster than you! 😆

Reply:
I have now moved/split some of the modules in a way that hopefully makes more sense for the dependency graph.  But, to my earlier point about #1043, I find this quite challenging to manage in my head without some kind of mechanical enforcement.  For me, at least, I think that maintaining optimal placement of the module files will only be practical once it is enforced/assisted by true package divisions.

For now, is the module arrangement sufficient to merge this PR, and we can perfect their placement later?